targets/pna_nic/thrift/pna_nic.thrift

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2024 Marvell Technology, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rupesh Chiluka (rchiluka@marvell.com)
+ *
+ */
+
+namespace cpp pnic_runtime
+namespace py pnic_runtime
+
+struct MirroringSessionConfig {
+  // 1:optional i32 port;
```

Comment:
This doesn't seem right to me
Why is port being removed?

Reply:
Yes, the port should be included. It can be either a network port or a host port.

In PSA, this port refers to `egress_port` for regular incoming packets and cloned packet (https://p4.org/p4-spec/docs/PSA.html#sec-clone), for the packet to be sent to a specific egress pipeline buffer.

In PNA, we don't have egress_port, and the cloned packets can be sent to either a host port or network port (https://p4.org/p4-spec/docs/PNA.html#sec-network-to-host-with-mirror-copy-to-different-host).

Currently, I have hardcoded the output port. I have yet to find out how to add output port in the main control / externs and how to differentiate between a host port and network port. For now, I commented out the port from the `MirroringSessionConfig`.

Reply:
Why not remove all the mirroring stuff from this PR, given that it is not implemented / used at the moment?

Reply:
`MirroringSessions` class is used to create sessions for both regular incoming packets and cloned packets. `MirroringSessions` class is also needed by a regular packet.

```
PnaNic::PnaNic(bool enable_swap)
  : Switch(enable_swap),
    input_buffer(1024),
    output_buffer(128),
    my_transmit_fn([this](port_t port_num, packet_id_t pkt_id,
                          const char *buffer, int len) {
        _BM_UNUSED(pkt_id);
        this->transmit_fn(port_num, buffer, len);
    }),
    pre(new McSimplePreLAG()),
    start(clock::now()),
    mirroring_sessions(new MirroringSessions()) {
  add_component<McSimplePreLAG>(pre);

  add_required_field("pna_main_parser_input_metadata", "recirculated");
  ...
  add_required_field("pna_main_output_metadata", "class_of_service");

  force_arith_header("pna_main_parser_input_metadata");
  force_arith_header("pna_main_input_metadata");
  force_arith_header("pna_main_output_metadata");
}
```

Reply:
I don't understand the point you are trying to make.
Currently the PNA implementation in this PR does *not* implement mirroring. There is no call to `mirroring_get_session ` in `targets/pna_nic/pna_nic.cpp`. Therefore, there is no value IMO in keeping the runtime functions used to configure mirroring, especially if they do not match the actual PNA architecture definition.